internal/job/git.go

```diff
@@ -130,18 +137,21 @@ func gitCleanSubmodules(ctx context.Context, sh *shell.Shell, gitCleanFlags stri
 	return nil
 }
 
-func gitFetch(
-	ctx context.Context,
-	sh *shell.Shell,
-	gitFlags, gitFetchFlags, repository string,
-	withRetry bool,
-	refSpec ...string,
-) error {
+type gitFetchArgs struct {
+	Shell         *shell.Shell // The shell to run the command in
+	GitFlags      string       // Global git flags to pass to the command
+	GitFetchFlags string       // Flags to pass to the fetch command
+	Repository    string       // The remote to fetch from
```

Comment:
i was a little surprised to see that in the `git fetch` command, this argument is indeed called `repository`, when we almost exclusively call it with `origin` (a remote), but that is apparently real.